references/cli/adopt.go

```diff
@@ -440,6 +441,19 @@ func NewAdoptCommand(f velacmd.Factory, streams util.IOStreams) *cobra.Command {
 		Run: func(cmd *cobra.Command, args []string) {
 			cmdutil.CheckErr(o.Complete(f, cmd, args))
 			cmdutil.CheckErr(o.Validate())
+			if o.AppName != "" {
+				app, err := f.AppGetter().Get(o.AppName)
+				if err == nil && app != nil {
+					if !o.Yes {
+						var confirm string
+						fmt.Fprintf(o.IOStreams.Out, "Application '%s' already exists, apply will override the existing app with the adopted one, please confirm [Y/n]: ", o.AppName)
+						fmt.Fscanf(o.IOStreams.In, "%s", &confirm)
+						if confirm != "Y" {
+							return
+						}
+					}
+				}
+			}
```

Comment:
you'd better add the feature inside the `(opt *AdoptOptions) Complete` function.